pyproject.toml

```diff
@@ -146,17 +144,9 @@ ignore = [
     # Missing docstring in `__init__`
     "D107",
 
-    # Manual list comprehension.
-    "PERF401",
     # Manual dict comprehension.
     "PERF403",
 
-    # Convert Dict[X] to dict[x], which we cannot do while we support Python 3.8
-    "UP006",
-
-    # Convert Union[X, Y] to X | Y, which we cannot do while we support Python 3.8
-    "UP007",
-
```

Comment:
Why is it that we can enable these now? They used to generate warnings because we use e.g. `Union[str, int]`, and we still do need to support Python 3.8, and we haven't added the `__future__` annotations import yet.

Reply:
I tried linting without these two rules and there were no warnings.

Reply:
ruff uses target-version setting and doesn't enable UP007 for us because:
- py38
- from future not used

I wonder if perhaps version elicitation was broken before or what?